TCAT/Controllers/FavoritesTableViewController.swift

```diff
@@ -149,28 +119,28 @@ extension FavoritesTableViewController: UITableViewDelegate {
         let cell = tableView.cellForRow(at: indexPath)
         cell?.accessoryView = UIActivityIndicatorView()
         tableView.deselectRow(at: indexPath, animated: true)
-        let place = resultsSection.items[indexPath.row]
-
-        if place.type == .busStop {
-            SearchTableViewManager.shared.insertPlace(for: Constants.UserDefaults.favorites, place: place, bottom: true)
-            dismissVC()
-        } else {
-            // Fetch coordinates and store
-            CoordinateVisitor.getCoordinates(for: place) { (latitude, longitude, error) in
-                if error != nil {
-                    print("Unable to get coordinates to save favorite.")
-                    cell?.accessoryView = nil
-                    let title = Constants.Alerts.PlacesFailure.title
-                    let message = Constants.Alerts.PlacesFailure.message
-                    let alertController = UIAlertController(title: title, message: message, preferredStyle: .alert)
-                    let done = UIAlertAction(title: Constants.Alerts.PlacesFailure.action, style: .default)
-                    alertController.addAction(done)
-                    self.present(alertController, animated: true, completion: nil)
-                } else {
-                    place.latitude = latitude
-                    place.longitude = longitude
-                    SearchTableViewManager.shared.insertPlace(for: Constants.UserDefaults.favorites, place: place, bottom: true)
-                    self.dismissVC()
+        if let place = resultsSection.getItem(at: indexPath.row) {
+            if place.type == .busStop {
+                Global.shared.insertPlace(for: Constants.UserDefaults.favorites, place: place, bottom: true)
+                dismissVC()
+            } else {
```

Comment:
we can add a `return` after line 125 and delete this `else` so that its not indented